applications/minotari_node/src/grpc/base_node_grpc_server.rs

```diff
@@ -319,6 +340,11 @@ impl tari_rpc::base_node_server::BaseNode for BaseNodeGrpcServer {
                         height: current_height,
                         timestamp: current_timestamp.as_u64(),
                         pow_algo: pow_algo.as_u64(),
+                        num_coinbases: coinbases.len() as u64,
+                        first_coinbase_extra: coinbases
+                            .first()
+                            .map(|c| c.features.coinbase_extra.to_vec())
+                            .unwrap_or_default(),
```

Comment:
- Why only the first coinbase extra?
- If so, how do we know the correct one is returned? Is the ordering not important?

Reply:
I think the ordering is important, but it's tricky. We allow multiple coinbases in a block, but that makes it difficult to put in extra merge mining data. It would be easier if the extra was in the header. Perhaps we should mark one coinbase and the primary coinbase, or only allow one coinbase to have an extra data field. @SWvheerden what do you think?

Reply:
I think perhaps a simpler solution might be a better fit here. 
We have the same problem with transactions and identifying them. They have multiple excess signatures, and we use one to identify them. 
I think here we do something similar. Rather than just returning the first coinbase extra, let's return the first coinbase with extra data in it. I think it will likely only have a single coinbase with extra data in it. There might be some cases where the miner wants to use more than one coinbase put in data, but we cant stop that really...

Reply:
ah good idea